portfolio/src/main/webapp/quiz.js

```diff
@@ -2,40 +2,125 @@
  * Question data type.
  */
 class Question {
-  constructor(question, answers, correct) {
+  /**
+   * @param {string} question The question that shall be posed
+   * @param {Array<string>} answers The potential responses to the question
+   * @param {number} correctAnswerIndex The index in answers that leads to the correct response
```

Comment:
nit - I don't know much about JSdoc but I recall seeing ! before Array before. Might be good to start thinking about documenting the [nullability of params](https://google.github.io/styleguide/jsguide.html#jsdoc-nullability).

Reply:
Ok, I will look into that.